geode-apis-compatible-with-redis/src/test/java/org/apache/geode/redis/internal/data/RedisHashTest.java

```diff
@@ -258,6 +258,13 @@ public void hscanSnaphots_shouldExpireAfterExpiryPeriod() {
     });
   }
 
+  @Test
+  public void hashSizeOverhead_shouldNotBeChanged_withoutForethoughtAndTesting() {
+    assertThat(RedisHash.PER_OBJECT_OVERHEAD).isEqualTo(8);
+    assertThat(RedisHash.getPerStringOverhead()).isEqualTo(RedisHash.PER_OBJECT_OVERHEAD + 46);
+    assertThat(RedisHash.getPerHashOverhead()).isEqualTo(RedisHash.PER_OBJECT_OVERHEAD + 116);
+  }
```

Comment:
Could we derive the overhead via the `ReflectionObjectSizer` here and compare that to the constant instead of hard coding what we derived?  Right now, someone could alter the code in a way that increases the overhead without changing the constant, but these tests wouldn't catch it.

Reply:
+1 to what Sarah said!

Reply:
we went to go write tests for this and found that we'd already done it in PartitionedRegionStatsUpdateTest. it's the last set of tests, divided by `/**` comments

Reply:
Are we looking at hash_bytesInUse_shouldReflectActualSizeOfDataInRegion and those tests? Maybe it would be better if those tests were actually computing the size of RedisHash, rather than HashMap<ByteArrayWrapper,ByteArrayWrapper>?

Reply:
Could we do some more direct unit testing in the `RedisHashTest.java`, maybe something like:

```
assertThat(ros.sizeof(new RedisHash())).isEqualTo(RedisHash.getPerHashOverhead());
assertThat(ros.sizeof(new RedisHash(createListOfDataElements(1))))
        .isEqualTo(RedisHash.getPerHashOverhead() + (RedisHash.getPerStringOverhead() * 2));
```

...and something similar in `RedisStringTest.java` and `RedisSetTest.java`

...it should be noted that the above assertions currently fail though, so maybe that's not the correct way to test it.  How did we originally derive the `PER_HASH_OVERHEAD` and `PER_STRING_OVERHEAD`? @upthewaterspout what is the proper way to derive/test those constants?

Reply:
This turns out to be hard. We can create a ReflectionObjectSizer in the constructor, and get the size of the object without any members added.

Then we'd have to examine at least one member being added, and attempt to derive the per-object overhead as we added it. Except that it turns out that an empty HashSet doesn't allocate some storage. When you add the first object, you add both the object and the internal storage. So we could then add a second object, and based on the length attempt to figure out what the 'internal storage overhead' is...

Anyway, it's not trivial. But we might have to do it because it looks like - no surprise and as feared - JVM v11 doesn't have the same overhead as v8.

Reply:
I was able to get the Sets to be dynamic - however, the values didn't change for Java 8, Java 11, and Liberica v8. Hashes are more complicated - the numbers just don't seem to add up, trying to figure things out from inside the constructor.